projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/func/LLVMCallUnboxNode.java

```diff
@@ -60,134 +47,135 @@
  */
 public abstract class LLVMCallUnboxNode {
 
-    @NodeChild(type = LLVMExpressionNode.class, value = "functionCallNode")
-    public abstract static class LLVMI1CallUnboxNode extends LLVMI1Node {
+    @NodeChild(type = LLVMDataNode.class, value = "functionCallNode")
+    public abstract static class LLVMI1CallUnboxNode extends LLVMDataNode {
 
         @Specialization
-        public boolean executeI1(Object value) {
-            return (boolean) value;
+        public boolean executeI1(boolean value) {
+            return value;
         }
     }
 
-    @NodeChild(type = LLVMExpressionNode.class, value = "functionCallNode")
-    public abstract static class LLVMI8CallUnboxNode extends LLVMI8Node {
+    @NodeChild(type = LLVMDataNode.class, value = "functionCallNode")
+    public abstract static class LLVMI8CallUnboxNode extends LLVMDataNode {
 
         @Specialization
-        public byte executeI8(Object value) {
-            return (byte) value;
+        public byte executeI8(byte value) {
+            return value;
         }
     }
 
-    @NodeChild(type = LLVMExpressionNode.class, value = "functionCallNode")
-    public abstract static class LLVMI16CallUnboxNode extends LLVMI16Node {
+    @NodeChild(type = LLVMDataNode.class, value = "functionCallNode")
+    public abstract static class LLVMI16CallUnboxNode extends LLVMDataNode {
 
         @Specialization
-        public short executeI16(Object value) {
-            return (short) (int) value;
+        public short executeI16(int value) {
+            return (short) value;
         }
     }
 
-    @NodeChild(type = LLVMExpressionNode.class, value = "functionCallNode")
-    public abstract static class LLVMI32CallUnboxNode extends LLVMI32Node {
+    @NodeChild(type = LLVMDataNode.class, value = "functionCallNode")
+    public abstract static class LLVMI32CallUnboxNode extends LLVMDataNode {
 
         @Specialization
-        public int executeI32(Object value) {
-            return (int) value;
+        public int executeI32(int value) {
+            return value;
```

Comment:
not really, however, if a function would return a result of a wrong types these nodes report a type error earlier.
Should I remove these nodes?